src/ray/raylet/node_manager.cc

```diff
@@ -1538,6 +1554,30 @@ void NodeManager::NewSchedulerSchedulePendingTasks() {
   DispatchScheduledTasksToWorkers();
 }
 
+void NodeManager::WaitForTaskArgsRequests(std::pair<ScheduleFn, Task>  &work) {
+  RAY_CHECK(new_scheduler_enabled_);
+  std::vector<ObjectID> object_ids = work.second.GetTaskSpecification().GetDependencies();
+  std::vector<ObjectID> required_object_ids;
+  for (auto const &object_id : object_ids) {
+    if (!task_dependency_manager_.CheckObjectLocal(object_id)) {
+      // Add any missing objects to the list to subscribe to in the task
+      // dependency manager. These objects will be pulled from remote node
+      // managers and reconstructed if necessary.
+      required_object_ids.push_back(object_id);
+    }
+  }
+
+  ray::Status status = object_manager_.Wait(
+      object_ids, -1, object_ids.size(), false,
```

Comment:
Shouldn't we be waiting on `required_object_ids` here?

Reply:
Hmm if the object manager is checking if they are local, we probably don't need the check above since it would be redundant.

Reply:
Yes, this was my understand, and that's why it doesn't check.